docs/decisions/0022-settings-simplification.rst

```diff
@@ -0,0 +1,298 @@
+      defaults,
+
+      * and which imports ``lms/envs/common.py``, which also sets
+        production-inappropriate defaults;
+
+    * it sets more defaults, some of the edX.org-specific;
+
+    * it loads ``/openedx/config/cms.yml``...
+
+      * which is generated by Tutor from template
+        ``tutor/templates/apps/openedx/config/cms.env.yml``
+
+        * which derives
+          ``tutor/templates/apps/openedx/config/partials/auth.yml``;
+
+    * it reverts some of ``/openedx/config/cms.yml`` with new "defaults";
+
+    * and it uses certain values ``/openedx/config/cms.yml`` to conditionally
+      override more settings and update certain dictionary settings, in a way
+      which is not documented.
+
+This is very difficult to reason about. Configuration complexity is frequently
+cited as a chief area of pain for Open edX developers and operators.
+Discussions in the Named Release Planning and Build-Test-Release Working Groups
+frequently are encumbered with confusion and uncertainty of what the default
+settings are in edx-platform, how they differ from Tutor's default settings,
+what settings can be overriden, and how to do so. Only a minority of developers
+and operators fully understand the configuration logic described above
+end-to-end; even for those that do, following this override chain for any given
+Django setting is time-consuming and error-prone. CAT-1 bugs and high-severity
+security vulnerabilities have arisen due to misunderstanding of how
+edx-platform Django settings are rendered.
+
+Developers are frequently instructed that if they need to override a Django
+setting, the preferred way to do so is to "make a Tutor plugin". This is a
+large amount of prior knowledge, boilerplate, and indirection, all required
+to simply do something which Django provides out-of-the-box via a custom
+``DJANGO_SETTINGS_MODULE``.
+
+Finally, it is worth nothing that all the complexity and toil exists alongside
+other edx-platform configuration methods, such as Waffle, configuration models,
+site configuration, XBlock configuration, and entry points. Those configuration
+pathways are outside of the scope of this ADR, but are mentioned to demonstrate
+the distressing level of complexity that developers and operators face when
+working with the platform.
+
+Decision
+********
+
+This is our target edx-platform settings module structure:
+
+* ``openedx/envs/common.py``: Defaults shared between LMS and CMS (new!).
+
+  * ``lms/envs/common.py``: LMS default settings. Wherever possible,
+    prod-ready values should be used; where not possible, obviously-incorrrect
+    values should be used.
+
+    * ``lms/envs/test.py``: Override LMS settings for unit tests. Should work
+      in a local venv as well as in CI.
+
+    * ``$THIRD_PARTY/lms/production.py``: Third-party providers (like edx.org) and
+      tools (like Tutor) will need to provide a custom setting settings file
+      derived from common.py in order to deploy LMS.
```

Comment:
I initially read this as we will allow 3rd parties to put their settings file in a 3rd party folder in the LMS, but I think that's not what you mean right? It might be worth clarifying that.

Reply:
Fair. Updated, LMK if it reads better.

Reply:
Made one more suggested improvement but we're in NIT territory so take or leave.

Reply:
Actually pushed this time, LMK what you think